ElectronClient/InteropServiceHelper.js

```diff
@@ -92,6 +95,16 @@ class InteropServiceHelper {
 		return this.exportNoteTo_('printer', noteId, options);
 	}
 
+	static async defaultFilename(noteIds, fileExtension, folders = null) {
+		const note = await Note.load(noteIds[0]);
+		if (folders) {
+			const folder = Folder.byId(folders, note.parent_id);
+			return friendlySafeFilename(`${note.title} - ${folder.title}.${fileExtension}`, 255);
+		} else {
+			return friendlySafeFilename(`${note.title}.${fileExtension}`, 255);
```

Comment:
I've just realised there's an issue with this code (which was already present in the previous version): the friendlySafeFilename and safeFilename functions are supposed to process the filename, *excluding the extension*. In other words it doesn't try to preserve the extension when truncating at 255 characters, which obviously can be a problem for long note titles. So the expected usage is `safeFilename(unsafeString) + "." + extension`.

Also default length is already 255 so you can remove this argument.

For the first case, I wonder why the folder title needs to be appended? And if it needs to be there, I would actually put it before like in a file path? Maybe @mic704b can clarify?

Reply:
> I wonder why the folder title needs to be appended

It was for when exporting multiple notes, in case multiple notes in different folders had the same title.

It would output files like:
eg `Tasks - Project 1.pdf`, `Tasks - Project 2.pdf`, `Tasks - Project 3.pdf`

Without it, you end up with multiple notes with the same name and a number appended. 
eg `Tasks.pdf`, `Tasks (2).pdf`, `Tasks (3).pdf`

It was for usability only, and is not a strict requirement.

Reply:
Looks like there were probably filename collisions when exporting multiple notes to pdf at the same time. I left the behavior in there but would happily remove it. Or change it if you have an idea for better behavior.

Reply:
Actually, I'm going to re-write this a bit and always export the containing notebook.

Reply:
Okay that was easier than I expected, should be good now

Reply:
@mic704b, thanks for clarifying and that indeed make sense. @CalebJohn, yes let's keep that behaviour but I've just added a comment that the friendlyFilename function should be applied twice to avoid cutting of the note title for particularly long folder titles.